### PR TITLE
[cleanup, ottool] Logging and user experience improvements

### PR DESCRIPTION
**Pull request by cfrantz**
1. Give `bootstrap` a progress bar.
2. Improve logging for `load-bitstream`; add a progress bar.
3. Log success or error for every command.

### CODE REVIEW

File: sw/host/opentitanlib/src/app/mod.rs [hunk 1]
```diff
@@ -15,11 +15,23 @@ use crate::transport::{ProxyOps, Transport, TransportError};
...
+/// Helper function to create a progress bar in the same form for each of
+/// the commands which will use it.
+pub fn progress_bar(total: u64) -> ProgressBar {
+    let progress = ProgressBar::new(total);
+    progress.set_style(
+        ProgressStyle::default_bar()
+            .template("[{elapsed_precise}] [{wide_bar}] {bytes}/{total_bytes} ({eta})"),
+    );
+    progress
+}
```
Comment (jrvanwhy) [inline sw/host/opentitanlib/src/app/mod.rs:33]: Unless I'm missing something, `progress_bar` is only used in opentitantool, not opentitan lib. Why was it moved to opentitanlib? Is there a reason why users of opentitanlib other than opentitantool would want to use the same progress bar configuration as opentitantool?

Comment (cfrantz) [reply sw/host/opentitanlib/src/app/mod.rs:33]: Yes: it was moved to opentitanlib::app, a collection of useful things you might want when building an application on top of opentitanlib.

This currently includes the `CommandDispatch` trait, the `TransportWrapper` struct and the pinmap/etc configuration structs.  This is probably in need of a refactor for better separation of concerns.

File: sw/host/opentitanlib/src/bootstrap/mod.rs [hunk 2]
```diff
@@ -116,6 +117,17 @@ impl<'a> Bootstrap<'a> {
...
+        Self::update_with_progress(transport, options, payload, |_, _| {})
+    }
+
+    /// Perform the update, sending the firmware `payload` to a SPI or UART target depending on
+    /// given `options`, which specifies protocol and port to use.
+    pub fn update_with_progress(
+        transport: &TransportWrapper,
+        options: &BootstrapOptions,
+        payload: &[u8],
+        progress: impl Fn(u32, u32),
```
Comment (a-will) [inline sw/host/opentitanlib/src/bootstrap/mod.rs:130]: Random: I just noticed this, but shouldn't our Rust style guide include doc comments for parameters of functions just like our C style guide?

**Review by a-will — APPROVED** [review]

**Review by jrvanwhy — APPROVED** [review]

File: sw/host/opentitanlib/src/app/mod.rs (hunk 1 above)
Comment (alphan) [reply sw/host/opentitanlib/src/app/mod.rs:33]: This could also be useful for e2e tests (if we have any tests that will take some time).

**Pull request merged**